integration/cluster.go

```diff
@@ -711,6 +711,15 @@ func (m *member) StopWithAutoLeaderTransfer(t *testing.T) {
 	plog.Printf("stopped %s (%s)", m.Name, m.grpcAddr)
 }
 
+// checkLeaderTransition waits for leader transition, returning the new leader ID.
+func checkLeaderTransition(t *testing.T, m *member, oldLead uint64) uint64 {
+	interval := time.Duration(m.s.Cfg.TickMs) * time.Millisecond
+	for m.s.Lead() == 0 || (m.s.Lead() == oldLead) {
```

Comment:
@heyitsanthony Yeah... Old leader sends time-out to follower, follower then `r.reset(r.Term + 1)` with `r.becomeCandidate()` losing its leader while campaigning. Any suggestion to differentiate between regular election and the transfer?